### DISCUSSION TITLE
Add watermark on multiple page of pdf with CLI

### DISCUSSION THREAD

**Discussion by mathias22osterhagen22** [ImageMagick/ImageMagick#2770] [Help]
Hello, 
I'm a newbie and I'm struggling a lot to add a "logo.png" rotate to -45 degree in the center of the pages of my PDF file.

Like this: 
![image](https://user-images.githubusercontent.com/60438474/97432549-6f2e0e00-191c-11eb-8316-da0a2753ad2c.png)

I tried with : 
 `convert  .\original.pdf null: .\logo.png -gravity center -rotate -45 -quality 100 -fill "rgba(180,180,180,0.80)" -pointsize 18 -weight bold -annotate +0+50 'Ihubtimespan' -compose multiply -layers composite destination.pdf`

But it rotates the entire PDF, + I figured out that the PDF size of the destination is around 6M (original is 84kb and logo is 70kb) is there a way to do it less big?

I found one old post on the magick forum but it is to convert from standard input: [https://imagemagick.org/discourse-server/viewtopic.php?f=1&t=35852&p=165102&hilit=watermark+pdf#p165102](https://imagemagick.org/discourse-server/viewtopic.php?f=1&t=35852&p=165102&hilit=watermark+pdf#p165102)

Thank you for all answear !

**Comment by snibgo** [comment]
You want to rotate just the annotated logo, so you need parentheses. Assuming bash syntax, they must be escaped with backslash:
```
convert .\original.pdf null: \( logo.png -gravity center -rotate -45 -quality 100 -fill "rgba(180,180,180,0.80)" -pointsize 18 -weight bold -annotate +0+50 'Ihubtimespan' \) -compose multiply -layers composite destination.pdf
```

Regarding the size, I suppose your PDF contains vectors and text. IM is a raster image processor, so it converts every page from the PDF to a raster image (an image with pixels). This is very much larger than vector text. And, of course, you will not be able to select text from the resulting PDF.

**Reply by mathias22osterhagen22** [reply]
Hello  @snibgo,

Thank you for your answer, I tried with the parentheses `()` with and without the escape backslash but the "logo" image is opened by the viewer of the window and I got an error message: 

**With the `\`:**
`convert .\original.pdf null: \( ./logo33.png -gravity center -rotate -45 -quality 100 -fill "rgba(180,180,180,0.80)" -pointsize 18 -weight bold -annotate +0+50 'Ihubtimespan' \) -compose multiply -layers composite destination.pdf`

***Result:***
>   convert.exe: unable to open image '\': No such file or directory @ error/blob.c/OpenBlob/3496.
>   convert.exe: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.
>   convert.exe: missing Null Image List Separator layers Composite @ error/mogrify.c/MogrifyImageList/8497.

**Without the `\`:**
` convert .\original.pdf null: ( ./logo33.png -gravity center -rotate -45 -quality 100 -fill "rgba(180,180,180,0.80)" -pointsize 18 -weight bold -annotate +0+50 'Ihubtimespan' ) -compose multiply -layers composite destination.pdf`

***Result:***
>   convert.exe: missing Null Image List Separator layers Composite @ error/mogrify.c/MogrifyImageList/8497.

Thank you for your time.

PS: I'm on W10

**Comment by snibgo** [comment]
What version of IM do you use, on what platform?

**Reply by mathias22osterhagen22** [reply]
Hello @snibgo,

There is the result of `magick -version` command:
```
Version: ImageMagick 7.0.9-22 Q16 x64 2020-02-08 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
```

I'm currently on `Windows 10 Pro x64bit`, hope this can help!

**Comment by snibgo** [comment]
Your command works for me, on Windows 8.1. What do you get from:
```
magick .\original.pdf info:
```
Perhaps you don't have Ghostscript.

**Reply by mathias22osterhagen22** [reply]
Pretty sure that I have it, could it be that magick doesn't find it?

![image](https://user-images.githubusercontent.com/60438474/98218540-27c30580-1f4c-11eb-8664-082c6b48b7de.png)

**Reply by fmw42** [reply]
Your delegates list gslib, so you have it.

`Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib`

But your policy.xml file may prevent you from using it. Check your policy.xml file to see if you have proper read privileges
